sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala

```diff
@@ -460,4 +460,29 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSparkSession {
       Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) ::
       Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) :: Nil)
   }
+
+  test("SPARK-34417 - test fillMap() for column with a dot in the name") {
+    val na = "n/a"
+    checkAnswer(
+      Seq(("abc", 23L), ("def", 44L), (null, 0L)).toDF("ColWith.Dot", "Col")
+        .na.fill(Map("`ColWith.Dot`" -> na)),
```

Comment:
It seems `Map("ColWith.Dot" -> na)` worked before but doesn't work now. I think it's a reasonable change but we should add an item in the migration guide `docs/sql-migration-guide.md`.